src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -218,7 +217,7 @@ public Optional<Map<Network, Set<PublicAddress>>> getPublicAddresses() {
          * A defensive copy of {@code publicAddresses} is used internally.
          */
         public void setPublicAddresses(Map<Network, Set<PublicAddress>> publicAddresses) {
-            this.publicAddresses = (publicAddresses != null) ? new HashMap<>(publicAddresses) : null;
+            this.publicAddresses = publicAddresses;
```

Comment:
This is very dangerous as publicAddress can now be changed from outside the class. Is there a reason for this change?

Reply:
Fixed in commit 39c5ce2c8be90a5c6695fc5696ac1f59b783b4e8